imageregistry/v1/types.go

```diff
@@ -0,0 +1,378 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	operatorsv1api "github.com/openshift/api/operator/v1"
+)
+
+const (
+	// DefaultRouteName is the name of the default route created for the registry
+	// when a default route is requested from the operator
+	DefaultRouteName = "default-route"
+
+	// ImageRegistryName is the name of the image-registry workload resource (deployment)
+	ImageRegistryName = "image-registry"
+
+	// ImageRegistryResourceName is the name of the image registry config instance
+	ImageRegistryResourceName = "cluster"
+
+	// ClusterProxyResourceName is the name of the cluster proxy config instance
+	ClusterProxyResourceName = "cluster"
+
+	// CloudCredentialsName is the name of the cloud credentials secret
+	CloudCredentialsName = "installer-cloud-credentials"
```

Comment:
Is it our public API? I'd say it a part of implementation details.

@bparees will we able to remove such constants later?

Reply:
yeah i was debating this myself...   once you put a constant in the api package it's part of the api, so you should not change/remove it after that.  If these are really just needed for internal code usage, i would not put them in api.

Reply:
I have removed the constants, it should not be part of the api as it is used only inside the operator. Thanks for catching this!